src/types.jl

```diff
@@ -41,11 +41,22 @@ type IVP{T,Y,F,G,J} <: AbstractIVP{T,Y}
     F!  ::F
     G!  ::G
     J!  ::J
+    reversed::Bool
```

Comment:
I think it's kinda awkward to have this here.  We decided that only `t0` but not `tend` belongs in this type.  I think the same can be said for including the time direction.  Move it to `Solver`?